src/python/WMCore/WMBS/MySQL/Files/SetLocationByLFN.py

```diff
@@ -11,14 +11,15 @@
 from WMCore.Database.DBFormatter import DBFormatter
 
 class SetLocationByLFN(DBFormatter):
-    sql = """INSERT INTO wmbs_file_location (fileid, location)
-             SELECT wmbs_file_details.id, wls.location
-               FROM wmbs_location_pnns wls, wmbs_file_details
-               WHERE wls.pnn = :location
-               AND wmbs_file_details.lfn = :lfn"""
+    sql = """INSERT INTO wmbs_file_location (fileid, pnn)
+                 SELECT wfd.id, wpnn.id
+                 FROM wmbs_pnns wpnn, wmbs_file_details wfd
+                 WHERE wpnn.pnn = :location
+                 AND wfd.lfn = :lfn"""
+
 
     def getBinds(self, lfn = None, location = None):
-        if type(lfn) == type('string'):
+        if isinstance(lfn, basestring):
             return {'lfn': lfn, 'location': location}
         elif isinstance(lfn, (list, set)):
             binds = []
```

Comment:
are we always expecting list of dictionary here? since you changed above/ In that case we'd better to remove location parameter.
Also we can just pass the list of dictionary instead of using this getBinds function.

Reply:
I might be wrong, but I think there are places passing a list of lfns to this dao, well, actually a list of dictionaries with lfn and location key/values. I can investigate to see if we can improve it, but honestly, I rather not change things at this stage, unless I really need to :-)

Reply:
I meant there is only cases which passes list of dict [{'lfn': xxx, 'location: uuu}, .....], I don't think it even works with list of lfns.  Also it seems getBinds is redundant in that case since lfn is already list of dict. I don't think there is any other place this is used than above code. (Didn't check the unittest).

Anyway, if you are concern about breaking the code. We can leave as it is and maybe change it later.